src/plugins/github/README.md

```diff
@@ -0,0 +1,175 @@
+# GitHub Plugin
+
+This is the SourceCred GitHub plugin. It creates a cred graph based on data
+from [GitHub].
+
+[GitHub]: https://github.com/
+
+## Nodes
+
+The GitHub plugin creates the following kinds of nodes:
+
+### Repo
+
+A GitHub repository, e.g. [sourcecred/sourcecred]. It will be directly
+connected to all of the pulls and issues in the repository. Setting a weight on
+the repository will have no effect (i.e. they do not mint cred). This will
+change when we switch to [CredRank].
+
+[CredRank]: https://github.com/sourcecred/sourcecred/issues/1686
+
+[sourcecred/sourcecred]: https://github.com/sourcecred/sourcecred
+
+### Issue
+
+A GitHub issue, e.g. [sourcecred/sourcecred#40]. Issues are connected to their
+author(s), to entities that they reference, to their comments, and to the
+containing GitHub repository. If issues are given a positive weight, then each
+issue will mint cred. Note that this could be abused by spammers.
+
+[sourcecred/sourcecred#40]: https://github.com/sourcecred/sourcecred/issues/40
+
+### Pull Request
+
+A GitHub pull request, e.g. [sourcecred/sourcecred#35][pull]. Pulls are
+connected to their author(s), to entities they reference, to their comments,
+their reviews, to their containing repository, and, (if merged) to the commit
+that they merged as. If pulls are given a positive weight, then each pull will
+mint cred, regardless of whether or not it merged. Note that this could be
+abused by spammers. There's a proposal ([merge-minted cred]) to have pulls mint
+cred only when they merge.
+
+[pull]: https://github.com/sourcecred/sourcecred/pull/35
+[merge-minted cred]: https://github.com/sourcecred/sourcecred/issues/1682
+
+### Pull Request Review
+
+A review of a GitHub pull request, e.g. [this review]. Reviews are connected to
+their author(s), to entities they reference, to their comments, and to the pull
+they review. If a positive weight is set on the reviews, then every review will
+mint cred; this could be abused by spammers.
+
+[this review]: https://github.com/sourcecred/sourcecred/pull/91#pullrequestreview-105254836
+
+### Comment
+
+A comment on an issue, pull request, or pull request review. Comments are
+connected to their author(s), to entities they reference, and to their 'parent'
+(the containing issue, pull, or review). If a positive weight is set on
+comments, then every comment will mint cred. This can be absued by spammers,
+and also may encourage long bikeshedding discussions or flame wars.
+
+### Commit
+
+A commit is a Git commit, as discovered via the GitHub API, e.g. [this commit].
+We currently enumerate every commit that is in the history of the `master`
+branch. We added commits back when we had a Git plugin; since we've disabled
+that plugin, commits are no longer well supported. Commits are connected to
```

Comment:
Agree here. Commit are still in the graph, so at first glance you could say. But the _are supported_, how are they "not well supported"?

I'm assuming by that you mean, they lack enough fidelity to be really useful in recognizing value. Where commits don't really add much value in themselves, the contents, authors, dependencies and touched files / directories do and we're only adding a few of those edges.

Reply:
I re-wrote this section to try to be clearer / more specific, let me know what you think

Reply:
I think what is throwing me off a bit is saying:

> We currently enumerate every commit that is in the history of the `master` branch.

to imply that we are doing something with commits, but then saying

> We added commits back when we had a Git plugin; since we've disabled that plugin, commits are no longer well supported.

And then I'm confused because we are in the README file _for_ the Git plugin, but the reference here is for "when we implement it"

> Commits will become useful once we re-implement a Git plugin

I think I'd suggest something along the lines of the following:

> A commit is a Git commit, as discovered via the GitHub API, e.g. [this commit]. Commits are connected to their author(s), to entities they reference, to their parent and child commits, and to the pull that merged them (if any). The original implementation of the plugin enumerated every commit in the history of the `master` branch. However, since commits are not well-supported (_what does this mean_?) they are currently not being used (_and is this for the implementation / version here, or for sourcecred's cred instance_?). When commits are re-implemented to (_fix what issues make them bad?_) then they can be added back to the plugin.

The above is rough (and where I don't know the answer I formatted with italics) but (based on my reading) it feels more correct to the state. I _think_ we want to say "This is what a commit is, this is why it's not a good implementation and is disabled, and this is how we can improve the implementation to be enabled (where again?)

that plugin, commits are no longer well supported. Commits are connected to